Library/Homebrew/keg.rb

```diff
@@ -483,11 +487,15 @@ def link_dir(relative_dir, mode)
       if src.symlink? || src.file?
         Find.prune if File.basename(src) == ".DS_Store"
         Find.prune if src.realpath == dst
-        # Don't link pyc files because Python overwrites these cached object
-        # files and next time brew wants to link, the pyc file is in the way.
+        # Don't link pyc or pyo files because Python overwrites these
+        # cached object files and next time brew wants to link, the
+        # file is in the way.
         if src.extname == ".pyc" && src.to_s =~ /site-packages/
           Find.prune
         end
+        if src.extname == ".pyo" && src.to_s =~ /site-packages/
+          Find.prune
+        end
```

Comment:
Can we combine these two checks and avoid the unnecessary (and possibly a bit too lenient) regular expression match? Here's what I have in mind:

```ruby
if %w[.pyc .pyo].include?(src.extname) && src.to_s.include?("/site-packages/")
  Find.prune
end
```